src/hiero_sdk_python/account/account_id.py

```
@@ -165,6 +283,67 @@ def to_string_with_checksum(self, client: "Client") -> str:
+            ) from e    
+        
+        try:
+            self.num = int(account_id.split(".")[-1])
+            return self
+        except (ValueError, AttributeError):
+            raise ValueError(f"Invalid account format received: {account_id}")
+
+    def populate_evm_address(self, client: "Client") -> "AccountId":
+        """Populate the EVM address using the Mirror Node."""
+        if self.num is None:
+            raise ValueError("Account number is required before populating evm_address")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)